sys/kern/tty.c

```diff
@@ -474,6 +475,31 @@ static void tty_in_hiwat(tty_t *tty) {
   }
 }
 
+static void tty_check_in_lowat(tty_t *tty) {
+  assert(mtx_owned(&tty->t_lock));
+
+  if (!(tty->t_flags & TF_IN_HIWAT))
+    return;
+
+  if (tty->t_inq.count < TTY_IN_LOW_WATER) {
+    tty->t_flags &= ~TF_IN_HIWAT;
+    tty_notify_in(tty);
+  }
+}
+
+static void tty_discard_input(tty_t *tty) {
+  assert(mtx_owned(&tty->t_lock));
+  ringbuf_reset(&tty->t_inq);
+  tty->t_line.ln_count = 0;
+  tty_check_in_lowat(tty);
+}
```

Comment:
Maybe is not a big feature/change, but code transfer in the same PR is not a good idea.

Reply:
Yeah, I've changed the code to declare the needed functions at the start of the file.